config/base/derive/src/lib.rs

```diff
@@ -37,19 +36,12 @@ pub fn load_from_disk_derive(input: TokenStream) -> TokenStream {
 }
 
 /// Derive for config querying and setting. More details in `iroha_config_base` reexport
-#[proc_macro_derive(Proxy)]
+#[proc_macro_derive(Proxy, attributes(config))]
```

Comment:
I don't understand why `config` attribute is required here

Reply:
<img width="986" alt="image" src="https://github.com/hyperledger/iroha/assets/43530070/d96d79af-3b66-4cb5-8205-f0d37e5b64f1">

Everything falls apart without it.

I didn't dive very deep into how configuration macros should be better implemented, because it will require deeper refactoring. Instead, I am trying to not touch them at all for now, because I am planning to remove them with the new config system further.

Adding `attributes(config)` here seems to be an optimal temporary solution.

Reply:
> I don't understand why config attribute is required here

It worked before because `derive(Documented)` consumed the `config` attribute.

It's a bit of a mess, because `rustc` stops generating the "cannot find attribute" error if any of the macros declare the attribute as supported, while in actuality all macros can observe all attributes. So, those config-related macros all used `config` attributes, but only `Documented` declared it as being used. The attribute should've been there from the beginning